packages/new-deploy/src/DeployContext.ts

```diff
@@ -35,12 +35,7 @@ export class DeployContext extends MultiProvider<config.Domain> {
   }
 
   get asNomadContext(): NomadContext {
-    const ctx = new NomadContext(this.data);
-
-    for (const [domain, provider] of this.providers.entries()) {
-      ctx.registerProvider(domain, provider);
-    }
-    return ctx;
+    return new NomadContext(this.data);
```

Comment:
this will break things no?

Reply:
i don't think so - registering provider is moved into the `NomadContext` constructor, so it's called there

Reply:
i suppose it's feasible that the DeployContext has other / more providers registered that aren't in the `data`?

Reply:
yes exactly. plus, signers aren't registered in the constructor

Reply:
https://github.com/nomad-xyz/monorepo/pull/154/commits/058b8788f36e8655af5fc7f559d3445f711ccdd5